File: Cpp/CXR_classify/storestage.cpp
#include "storestage.h"

StoreStage::StoreStage(ConfigHandler * configHandler, DatabaseHandler * dbHandler) : Stage()
{
    storer = new Storer(configHandler, dbHandler);
}

void StoreStage::store()
{
    threadpool->start(storer);
}

File: Go/CxrClassify/storeStage/storeStage.go
package storeStage

import (
	"CxrClassify/configHandler"
	"CxrClassify/databaseHandler"
	"CxrClassify/stage"
)

type StoreStage struct {
	stage.Stage

	Storer *Storer

	_ func() `constructor:"init"`
}

func (s *StoreStage) init() {

}

func (s *StoreStage) Setup(configHandler *configHandler.ConfigHandler, databaseHandler *databaseHandler.DatabaseHandler) {
	s.Storer = NewStorer(nil)
	s.Storer.Setup(configHandler, databaseHandler)
}

func (s *StoreStage) Store() {
	s.Storer.Run()
}
